lib/init.go

```diff
@@ -0,0 +1,221 @@
+					_, err := f.WriteString(checkStr(value.String(), val) + "\n\n") //write properties of subsections
+					if err != nil {
+						log.Fatal(err)
+					}
+					fmt.Println(key2.String())
+					if isArray(checkType(value.String())) == true {
+						f.WriteString("]")
+					}
+					return true
+				})
+			}
+			if key1.String() == "items" {
+				i := 0
+				result1 := gjson.GetBytes(content, "properties."+key.String()+".items.properties")
+				result1.ForEach(func(key2, value gjson.Result) bool {
+					if i != 0 {
+						f.WriteString(",")
+					}
+					i++
+					f.WriteString("\"" + key2.String() + "\":")
+					val := checkFormat(value.String())
+
+					if isArray(checkType(value.String())) == true {
+						f.WriteString("[")
+						_, err := f.WriteString(checkStr(value.String(), val)) //write items of subsections
+						if err != nil {
+							log.Fatal(err)
+						}
+					} else {
+						_, err := f.WriteString(checkStr(value.String(), val) + "\n\n") //write items of subsections
+						if err != nil {
+							log.Fatal(err)
+						}
+					}
+					fmt.Println(value.String())
+					if isArray(checkType(value.String())) == true {
+						f.WriteString("]")
+					}
+					return true
+				})
+			}
+
+			return true
+		})
+		f.WriteString("}")
+		if (isArray(checkType(value.String()))) == true {
+			_, err = f.WriteString("]")
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		return true
+	})
+	f.WriteString("}")
+	f.Close()
+
+	var schema schema.Data
+	if err := json.Unmarshal(content, &schema); err != nil {
+		log.Fatal(err)
+	}
+
+	tpl, err := template.ParseFiles(tmpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	file, err := os.Create(doc)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	if err = tpl.Execute(file, schema); err != nil {
+		log.Fatal(err)
+	}
+	err = os.Remove(tmpl)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+}
+
+func checkStr(x, typ string) string {
+	d := "\"description\": "
+	var val string
+	print(x)
+	if strings.Contains(x, d) {
+		s := make([]string, 0, 2)
+		s = strings.Split(x, d)
+		if typ == "date" {
+			val = "\"" + extractDate(x) + "\""
+			return val
+		}
+		if typ == "email" {
+			s = strings.Split(s[1], "e.g. ")
+			s = strings.Split(s[1], ",\n")
+			val = "\"" + s[0]
+			return val
+		}
+
+		if typ == "uri" {
+			pat := regexp.MustCompile(`https?://.*\.com`)
+			s := "\"" + pat.FindString(x) + "\""
+			return s
+		}
+
+		if typ == "location" {
+			str := "{\n \"lat\": \"0\",\n\"long\": \"0\"\n}"
+			return str
+		}
+
+		s = strings.Split(s[1], "\n")
+		val = s[0]
+
+		val = strings.TrimSuffix(val, ",")
+		return val
+	}
+	return "\"\""
+}
+
+func checkFormat(x string) string {
+	val := gjson.Get(x, "format")
+	if !val.Exists() {
+		return ""
+	}
+	return val.String()
+}
+
+func extractDate(str1 string) string {
+	re := regexp.MustCompile(`\d{4}-\d{2}-\d{2}`)
+	//re.MatchString(str1)) // true
+	match := re.FindStringSubmatch(str1)
+	return match[0]
+}
+
+func checkType(x string) string {
```

Comment:
Why checkType and not hasType or getType as it actually returns the value of type and not true/false.